pkg/node/manager/manager.go

```diff
@@ -404,15 +521,39 @@ func (m *manager) backgroundSync(ctx context.Context) error {
 		case <-startWaiting:
 		}
 
-		hr := m.health.NewScope("background-sync")
 		if err != nil {
-			hr.Degraded("Failed to apply node validation", err)
+			health.Degraded("Failed to apply node validation", err)
 		} else {
-			hr.OK("Node validation successful")
+			health.OK("Node validation successful")
 		}
 	}
 }
 
+const (
+	upAndRunningFlags    = unix.IFF_UP | unix.IFF_RUNNING
+	slaveOrLoopbackFlags = unix.IFF_SLAVE | unix.IFF_LOOPBACK
+)
+
+// Check if a given device is up and running (not down or carrier down).
+// Always returns false for non-physical devices so we never detect a
+// transition from down to up.
+func deviceIsUpAndRunning(dev *tables.Device) bool {
+	// We only care about physical devices. So we select for [netlink.Device].
+	// This technically also include loopback devices, but those are filtered
+	// in a later step.
+	if dev.Type != "device" {
+		return false
+	}
+
+	slaveOrLoopback := dev.RawFlags&slaveOrLoopbackFlags != 0
```

Comment:
Why do we filter slave devices here? Would be nice to have some explanation with the concrete scenario.

Reply:
I stole this bit of logic from devicesController.isSelectedDevice, mostly for the loopback check. And while I don't think it hurts, I have no specific need for the slave check. I will remove it.